app/models/dossier.rb

```diff
@@ -1148,6 +1148,18 @@ def termine_and_accuse_lecture?
     procedure.accuse_lecture? && termine?
   end
 
+  def track_can_passer_en_construction
+    if !revision.ineligibilite_enabled
+      yield
+      [true, true] # without eligibilite rules, we never reach dossier.champs.visible?, don't cache anything
+    else
+      from = can_passer_en_construction? # with eligibilite rules, self.champ[x].visible is cached by passing thru conditions checks
+      yield
+      champs.map(&:reset_visible) # si we we must reset self.champs[x].visible?, because an updated occured and we should re-evaluate champs[x] visibility
```

Comment:
```suggestion
      champs.map(&:reset_visible) # we must reset self.champs[x].visible?, because an update occurred and we should re-evaluate champs[x] visibility
```